Cloudwatch: Add Support for external ID in assume role (#23685)
grafana/grafana: docs/sources/features/datasources/cloudwatch.md

| _Auth Provider_            | Specify the provider to get credentials.                                                                |
| _Credentials_ profile name | Specify the name of the profile to use (if you use `~/.aws/credentials` file), leave blank for default. |
| _Assume Role Arn_          | Specify the ARN of the role to assume                                                                   |

> | _External ID_              | If assuming a role in another account, specify its ID.                     |

Reviewer: Unclear what "its" ID is. Please clarify.

Author: @oddlittlebird , @aknuds1 ,
How about this 
"If role to assume is in another account and role is created with External ID specify it."

Reviewer: Still unclear what "it" is. I was thinking something like what is below, assuming it is correct. From the context, it is unclear if the ID is for the role or the account.

If assuming a role in another account, specify the ID of the other account.

Author: @oddlittlebird ,
Thanks, i will make it. 

"If assuming a role in another account, specify the External ID of the role."

Reviewer 2: `External ID` is optional. It sounds like it is mandatory.

Author: Thanks @jangaraj ,
Changed : 
"Specify the External ID If assuming a role in another account and role is created with External ID."

Reviewer: I think the "if" at the beginning makes it clear that it is optional. If you start the sentence with "specify" so that it follows the same pattern as the other required options, that could be confusing.

I recommend, "If you are assuming a role in another account, then specify the External ID of the role."

Author: "External ID" is required only if  we are assuming a role in another account and if role is created with "External ID". Jangaraj  pointed out second condition  **In other account one can create role without "External ID"** 
Combining @oddlittlebird  and  @jangaraj suggestion. 
 
"If you are assuming a role in another account and if role is created with External ID, then specify the External ID. "

@jangaraj ,
Could you please review PR (first ever on github :) ) if i missed anything else.

Reviewer 2: "If you are assuming a role in another account and if role is created with External ID, then specify the External ID." looks good. It will be useful also in the field tooltip, because users without External ID can be confused if they need it or not.

Author: Sure, incorporated suggested changes.

> | _External ID_              | Specify the External id if you assume a role in another account having external ID.                     |

Reviewer 3: ```suggestion
| _External ID_              | If assuming a role in another account, specify its ID.                     |
```
